hopp/tools/design/wind/turbine_library_interface_tools.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+
+import PySAM.Windpower as windpower
+from turbine_models.parser import Turbines
+import hopp.tools.design.wind.power_curve_tools as curve_tools
+from hopp.utilities.log import hybrid_logger as logger
+import hopp.simulation.technologies.wind.floris as floris_wrapper
+
+def extract_power_curve(turbine_specs: dict, model_name: str):
+    """Creates power-curve for turbine based on available data and formats it for the corresponding simulation model.
+
+    Args:
+        turbine_specs (dict): turbine specs loaded from turbine-models library.
+        model_name (str): wind simulation model, either "pysam" or "floris".
+
+    Raises:
+        ValueError: if turbine data doesn't have the minimum required power-curve information.
+        ValueError: if model name is not either 'pysam' or 'floris'
+
+    Returns:
+        dict: power-curve dictionary formatted for the corresponding ``model_name``.
+    """
+
+    if model_name not in ("floris", "pysam"):
+        raise ValueError(f"model_name {model_name} is invalid, options are either 'floris' or 'pysam'.")
+    turbine_specs["power_curve"] = turbine_specs["power_curve"].dropna()
+    wind_speeds = np.nan_to_num(turbine_specs["power_curve"]["wind_speed_ms"].to_list())
+    turbine_curve_cols = turbine_specs["power_curve"].columns.to_list()
+    
+    has_cp_curve = "cp" in turbine_curve_cols
+    has_power_curve = "power_kw" in turbine_curve_cols
+    has_ct_curve = "ct" in turbine_curve_cols
+    
+    if not has_cp_curve and not has_power_curve:
+        turbine_name = turbine_specs["name"]
+        raise ValueError(f"turbine {turbine_name} does not have minimum required power curve data (needs either power_kw or cp)")
+    
+    if has_cp_curve:
+        cp_curve = np.array(turbine_specs["power_curve"]["cp"].to_list())
+        cp_curve = np.nan_to_num(cp_curve)
+        cp_curve = np.where(cp_curve<0,0,cp_curve).tolist()
+
+    if has_power_curve and has_cp_curve:
+        power_curve_kw = np.array(turbine_specs["power_curve"]["power_kw"].to_list())
+        power_curve_kw = np.nan_to_num(power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw<0,0,power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw>turbine_specs["rated_power"],turbine_specs["rated_power"],power_curve_kw).tolist()
+        
+        cp_curve = np.array(turbine_specs["power_curve"]["cp"].to_list())
+        cp_curve = np.nan_to_num(cp_curve)
+        cp_curve = np.where(cp_curve<0,0,cp_curve).tolist()
```

Comment:
This logic (and the same three lines below) are not needed because they've already been ran in the above code block for `if has_cp_curve` (L38). Delete these blocks?